src/utils/auth.go

```diff
@@ -68,12 +79,33 @@ func Authenticate(pc **gophercloud.ProviderClient, service string, config map[st
 	tlsConfig := &tls.Config{InsecureSkipVerify: tlsVerify}
 	transport := http.DefaultTransport.(*http.Transport).Clone()
 	transport.TLSClientConfig = tlsConfig
-	clientOpts.HTTPClient = &http.Client{Transport: transport}
 
-	*pc, err = clientconfig.AuthenticatedClient(&clientOpts)
+	ao, err := clientconfig.AuthOptions(&clientOpts)
+	if err != nil {
+		return fmt.Errorf("failed to build auth options: %w", err)
+	}
+
+	*pc, err = openstack.NewClient(ao.IdentityEndpoint)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create a provider: %w", err)
 	}
+	(*pc).HTTPClient.Transport = transport
+
+	// enable API debug logs
+	if debug, _ := strconv.ParseBool(GetEnv("OS_DEBUG", "false")); debug {
+		(*pc).HTTPClient.Transport = &client.RoundTripper{
+			Rt: transport,
+			Logger: osDebugger{log.WithFields(logrus.Fields{
+				"level": "os_debug",
```

Comment:
Here I would suggest to add fields that would help indicate that the log is from Openstack and also to indicate from which openstack service it is (swift, cinder, manila, ...). 

For example something like this would make it easier to search logs by json fields (in logging system) or using `kubectl logs`:
```go
			"component": service,
			"source":    "openstack",
```

Since each service has unique client, this should work fine. If you have better ideas than key names `component` and `source`, please suggest :-).

Reply:
fixed